CMakeLists.txt

```diff
@@ -141,6 +141,7 @@ option(FLB_WINDOWS_DEFAULTS    "Build with predefined Windows settings"     Yes)
 option(FLB_WASM                "Build with WASM runtime support"            Yes)
 option(FLB_WAMRC               "Build with WASM AOT compiler executable"    No)
 option(FLB_WASM_STACK_PROTECT  "Build with WASM runtime with strong stack protector flags" No)
+option(FLB_GZIP_LEVEL          "Set default gzip compression level")
```

Comment:
Agree. I *think* that's what @edsiper was getting at here:

> Have a default at build time (proposed solution), but use direct numbers instead of an internal macro (despite it could be the same).

That'll tidy up the build-time portion, which just leaves the runtime implementation. Hoping to knock it all out this weekend.

Reply:
Sounds like we're all in sync then and cheers for the effort @nallenscott 
It'll probably simplify the CMake setup as well.